### ISSUE TITLE
Bouncy Castle Schnorr Implementation

### ISSUE THREAD

**Issue by msgilligan** [bitcoinj/secp256k1-jdk#21]
The initial goal for the Bouncy Castle Schnorr Implementation should be for the Schnorr.{java,kt} examples to work properly.

Update: Now that we have unit tests, it makes sense to get those working first. Especially once the BIP340 test vector unit test is completed and merged.

**Comment by msgilligan** [comment]
This is now a subtask of https://github.com/bitcoinj/bitcoinj/issues/3389

**Comment by msgilligan** [comment]
This should include a Bouncy Castle implementation of `secp.xOnlyPubKeyParse(byte[] inputData)` (PR #320 adds the method with an FFM implementation and the Bouncy Castle implemntation throwing `UnsupportedOperationException`)

**Comment by msgilligan** [comment]
For reference, there is a now-closed issue on `bcgit/bc-java` requesting Schnorr Signature support:

* https://github.com/bcgit/bc-java/issues/285

When our implementation is ready, we should probably post a link to it on that issue.

There is also this issue that @schildbach opened:

* https://github.com/bcgit/bc-java/issues/1114

**Comment by Polve** [comment]
"now closed" since  Feb 4, 2018 :-)

**Comment by msgilligan** [comment]
> "now closed" since Feb 4, 2018 :-)

Yeah, I don't think Bouncy Castle is going to add Schnorr Signatures  any time soon. But we're really close to having Schnorr Signatures using Bouncy Castle types and EC math in this library.

**Comment by msgilligan** [comment]
We have a preliminary implementing in PR #369
